pkg/playback/http_playback.go

```diff
@@ -0,0 +1,413 @@
+		}
+		if err != nil {
+			fmt.Println("Error in /playback/mode handler: ", err)
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		fmt.Println("/playback/mode: Set mode to ", strings.ToUpper(modeString))
+	})
+
+	customMux.HandleFunc("/playback/cassette/setroot", func(w http.ResponseWriter, r *http.Request) {
+		const queryKey = "dir"
+		directoryVals, ok := r.URL.Query()[queryKey]
+
+		if !ok {
+			writeErrorToHTTPResponse(w, fmt.Errorf("\"%v\" query param must be present", queryKey), 400)
+			return
+		}
+
+		if len(directoryVals) > 1 {
+			fmt.Printf("Multiple \"value\" param values given, ignoring all except first: %v\n", directoryVals[0])
+		}
+
+		// directoryVal can be either a relative or absolute path - filepath.Abs() no-ops if the given path is already absolute, and converts to absolute (assuming CWD) if relative
+		absoluteCassetteDir, err := filepath.Abs(directoryVals[0])
+
+		if err != nil {
+			fmt.Println("Error with given directory in /playback/cassette/setroot handler: ", err)
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		err = rr.setCassetteDir(absoluteCassetteDir)
+
+		if err != nil {
+			fmt.Println("Error in /playback/cassette/setroot handler: ", err)
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+		// TODO: why is this not printing out the absolute path?
+		fmt.Printf("Cassette directory set to \"%v\"\n", rr.cassetteDirectory)
+		w.WriteHeader(200)
+	})
+
+	customMux.HandleFunc("/playback/cassette/load", func(w http.ResponseWriter, r *http.Request) {
+		filepathVals, ok := r.URL.Query()["filepath"]
+
+		if !ok {
+			err := fmt.Errorf("\"filepath\" query param must be present")
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		if len(filepathVals) > 1 {
+			fmt.Printf("Multiple \"filepath\" param values given, ignoring all except first: %v\n", filepathVals[0])
+		}
+
+		relativeFilepath := filepathVals[0]
+
+		if !strings.HasSuffix(strings.ToLower(relativeFilepath), ".yaml") {
+			err := fmt.Errorf("%v is not a .yaml file", relativeFilepath)
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		if filepath.IsAbs(relativeFilepath) {
+			err := fmt.Errorf("%v must be a relative filepath. a absolute filepath was provided", relativeFilepath)
+			writeErrorToHTTPResponse(w, err, 400)
+			return
+		}
+
+		err := rr.loadCassette(relativeFilepath)
+
+		if err != nil {
+			writeErrorToHTTPResponse(w, err, 500)
+			return
+		}
+
+		var isRecording bool
+		switch rr.mode {
+		case Record:
+			isRecording = true
+		case Replay:
+			isRecording = false
+		case Auto:
+			isRecording = rr.isRecordingInAutoMode
+		default:
+			writeErrorToHTTPResponse(w, errors.New("in unexpected mode state in handler. Please restart the playback server"), 500)
+			return
+		}
+
+		var statusMsg string
+
+		if isRecording {
+			statusMsg = fmt.Sprintf("Recording to %v", relativeFilepath)
+		} else {
+			statusMsg = fmt.Sprintf("Replaying from %v", relativeFilepath)
+		}
```

Comment:
(non-blocking) this bit could be its own method

Reply:
DRY'ed up in 961d13ecaf1f0c8830e80fe070101dbe0ce126ab